content/testing.md

```diff
@@ -0,0 +1,683 @@
+    FlowRouter.getParam.restore();
+    Meteor.subscribe.restore();
+  });
+
+  it('renders correctly with simple data', () => {
+    Factory.create('list', { _id: listId });
+    const timestamp = new Date();
+    const todos = _.times(3, i => Factory.create('todo', {
+      listId,
+      createdAt: new Date(timestamp - (3 - i)),
+    }));
+
+    withRenderedTemplate('Lists_show_page', {}, el => {
+      const todosText = todos.map(t => t.text).reverse();
+      const renderedText = $(el).find('.list-items input[type=text]')
+        .map((i, e) => $(e).val())
+        .toArray();
+      chai.assert.deepEqual(renderedText, todosText);
+    });
+  });
+});
+```
+
+Of particular interest in this test is the following:
+
+<h4 id="simple-integration-test-importing">Importing</h4>
+
+As we'll run this test in the same way that we did our unit test, we need to `import` the relevant modules under test in the same way that we [did in the unit test](#simple-integration-test-importing).
+
+<h4 id="simple-integration-test-stubbing">Stubbing</h4>
+
+As the system under test in our integration test has a larger surface area, we need to stub out a few more points of integration with the rest of the stack. Of particular interest here is our use of the [`stub-collections`](#mocking-the-database) and of [Sinon](https://sinonjs.org) to stub out Flow Router and our Subscription.
+
+<h4 id="simple-integration-test-data">Creating data</h4>
+
+In this test, we used [Factory package's](#generating-data) `.create()` API, which inserts data into the real collection. However, as we've proxied all of the `Todos` and `Lists` collection methods onto a local collection (this is what `stub-collections` is doing), we won't run into any problems with trying to perform inserts from the client.
+
+This integration test can be run the exact same way as we ran [unit tests above](#running-unit-tests).
+
+<h3 id="full-app-integration-test">Full-app integration test</h3>
+
+In the Todos example application, we have a integration test which ensures that we see the full contents of a list when we route to it, which demonstrates a few techniques of integration tests:
+
+```js
+/* eslint-env mocha */
+
+import { Meteor } from 'meteor/meteor';
+import { Tracker } from 'meteor/tracker';
+import { DDP } from 'meteor/ddp-client';
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import { assert } from 'meteor/practicalmeteor:chai';
+import { $ } from 'meteor/jquery';
+
+import { generateData } from './../../api/generate-data.app-tests.js';
+import { Lists } from '../../api/lists/lists.js';
+import { Todos } from '../../api/todos/todos.js';
+
+// Utility function to wait for subscriptions
+const waitForSubscriptions = (done) => {
+  const poll = Meteor.setInterval(() => {
+    if (DDP._allSubscriptionsReady()) {
+      clearInterval(poll);
+      done();
+    }
+  }, 200);
+};
+
+// Utility to allow throwing inside callback
+const catchAsync = (done, cb) => () => {
```

Comment:
Just making sure if you throw from inside async code it actually gets registered by the test driver (it's weird that you have to do this). See my promise note below